src/middleware.ts

```diff
@@ -0,0 +1,18 @@
+import type { NextRequest } from "next/server";
+import { auth0 } from "./lib/auth0";
+
+export async function middleware(request: NextRequest) {
```

Comment:
We do have public pages, such as `/auth`, `/community` etc. Will they be impacted by this?